Migration to gatsby v5 fresh start

sharnik opened the pull request:
This is based off work started in https://github.com/brains-and-beards/open-brainsandbeards-com/pull/8. 

I've created a new Gatsby v5 TypeScript project and started porting components from Szymon's branch and fixing any issues that came up. If you want to track the progress of this process, you can do that in [a dedicated temporary repo](https://github.com/brains-and-beards/brains-and-beards-com-dupla). Currently I think it's ready to be merged as it's basically as functional as current version on brainsandbeards.com.

There are some more tweaks (and bugfixing) that we need to do, but those can be done separately. Full list is in our Notion, dropping a screenshot here for context:

<img width="738" alt="CleanShot 2023-04-18 at 20 38 12@2x" src="https://user-images.githubusercontent.com/987/232872490-93555559-3ae7-443d-bbc5-fa4ebec78328.png">

It'd be great if you could test out the deployed PR preview whether there are any bugs that have not been found / tracked yet.

szymonkoper committed:
WIP

szymonkoper committed:
WIP

szymonkoper committed:
Adds omitKeys to gatsby-plugin-react-svg config

szymonkoper committed:
Deps update and other changes

szymonkoper committed:
Fixes ReadMore styles

szymonkoper committed:
Removed warnings on home pge

szymonkoper committed:
Progress on Our Clients

szymonkoper committed:
Displays clients logos

szymonkoper committed:
Adds sharoo case study

szymonkoper committed:
Updated SEO to Gatsby v5 requirements

szymonkoper committed:
Replaces gatsby-plugin-favicon with gatsby-plugin-manifest

szymonkoper committed:
Displays prettier svgs in layout

szymonkoper committed:
Updates team page

szymonkoper committed:
Updates message-sent page

szymonkoper committed:
Migration from gatsby-transformer-remark to gatsby-plugin-mdx

szymonkoper committed:
Displays blog posts list

szymonkoper committed:
Passes context data to blog post page

szymonkoper committed:
Showing blog post page

szymonkoper committed:
Adds canonical link for SEO purposes

szymonkoper committed:
Restored all blog posts

szymonkoper committed:
Restored showing of gifs

szymonkoper committed:
Restores (partially) code blocks in mdx

szymonkoper committed:
Replaces gatsby-remark-gifs with gatsby-remark-copy-linked-files

sharnik committed:
Update yarn lockfile

sharnik committed:
Settle on a single (Netlify) redirect setup

sharnik committed:
Move everything to a new Gatsby v5 TS project

sharnik committed:
Improve .gitignore

szymonkoper commented on src/components/shared/layout/Navbar.tsx:
```diff
@@ -0,0 +1,138 @@
+                { url: "/projects", label: "Projects" },
+                { url: "/about-us", label: "About us" },
+                { url: "/team", label: "Team" },
+                { url: "/jobs", label: "Careers" },
```
Something wrong with this redirect

> 404 Not found
> You just hit a route that doesn't exist... the sadness.

szymonkoper replied:
I went through all other links, including project case studies and they all look good apart from this "Careers" one. 👍

I have not checked blog posts yet

sharnik replied:
I've extracted the navbar link rendering to a separate component. Now it works as expected, the code is a bit cleaner, and I learnt what's going on there. (Apparently Gatsby has special JS handling for Link elements and treats them as internal.)

szymonkoper commented on README.md:
```diff
@@ -1,14 +1,49 @@
-# Brains & Beards web page
```
I think we should keep old `README.md` contents instead of initial one from new Gatsby project

sharnik replied:
I've updated it 👍

szymonkoper commented on package.json:
```diff
@@ -1,57 +1,52 @@
+  },
+  "dependencies": {
+    "@mdx-js/react": "^2.3.0",
+    "gatsby": "^5.7.0",
```
The newest is now 5.9.0 🌝

sharnik replied:
I'll take a break from updating Gatsby and leave it as-is :)

szymonkoper commented on src/pages/team.tsx:
```diff
@@ -1,93 +1,97 @@
               ))}
             </div>
             <div className="mobile-only team-swiper">
               <Swiper
```
On "Team" page on smartphone I see only Patryk.

In fact it shows a swiper, but not centered and without any arrows indicating it's possible to swipe it. 

<img width="512" alt="Screenshot 2023-04-19 at 10 02 38" src="https://user-images.githubusercontent.com/2790570/233009717-0cdcade7-5357-4574-b72a-bcdd25b03a72.png">

sharnik replied:
Fixed 👍

sharnik committed:
Fix team members display on small screens

sharnik committed:
Updated README

sharnik committed:
Add Prettier and update Netlify plugin

sharnik committed:
Global prettier formatting

sharnik committed:
Fix career page redirects

sharnik committed:
Better handling of Navbar links

We can only create _internal_ links as Link, everything else
(onClick elements, links that go to external sites, etc.) need
to use _<a/>_.

sharnik committed:
Cleaner separation between reference swiper items
